src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -503,3 +510,174 @@ def _sql_alter_from(self, table: Table, target_table_key: str, ws_id: int):
     def _is_migrated(self, schema: str, table: str) -> bool:
         index = self._migration_status_refresher.index()
         return index.is_migrated(schema, table)
+
+
+class ACLMigrator:
+    def __init__(
+        self,
+        tables_crawler: TablesCrawler,
+        grant_crawler: GrantsCrawler,
+        workspace_info: WorkspaceInfo,
+        backend: SqlBackend,
+        group_manager: GroupManager,
+        migration_status_refresher: MigrationStatusRefresher,
+        principal_acl: PrincipalACL,
+    ):
+        self._tc = tables_crawler
+        self._gc = grant_crawler
+        self._backend = backend
+        self._workspace_info = workspace_info
+        self._group = group_manager
+        self._migration_status_refresher = migration_status_refresher
+        self._principal_acl = principal_acl
+
+    def migrate_acls(
+        self,
+        *,
+        target_catalog: str | None = None,
+        legacy_table_acl: bool = True,
+        principal: bool = True,
+        hms_fed: bool = False,
+    ) -> None:
+        acl_strategies = []
+        workspace_name = self._workspace_info.current()
```

Comment:
```suggestion
        if not target_catalog: target_catalog = self._workspace_info.current()
```
we only need to compute current workspace name if target catalog is not supplied